includes/functions/functions_email.php

```diff
@@ -119,50 +112,61 @@ function zen_mail($to_name, $to_address, $email_subject, $email_text, $from_emai
             // ensure the address is valid, to prevent unnecessary delivery failures
             if (!zen_validate_email($to_email_address)) {
                 $zco_notifier->notify('NOTIFY_EMAIL_ADDRESS_VALIDATION_FAILURE', sprintf(EMAIL_SEND_FAILED . ' (failed validation)', $to_name, $to_email_address, $email_subject));
-                @error_log(sprintf(EMAIL_SEND_FAILED . ' (failed validation)', $to_name, $to_email_address, $email_subject));
+                error_log(sprintf(EMAIL_SEND_FAILED . ' (failed validation)', $to_name, $to_email_address, $email_subject));
                 continue;
             }
 
             //define some additional html message blocks available to templates, then build the html portion.
             if (is_array($block)) {
-                if (!isset($block['EMAIL_TO_NAME']) || $block['EMAIL_TO_NAME'] == '') {
+                if (empty($block['EMAIL_TO_NAME'])) {
                     $block['EMAIL_TO_NAME'] = $to_name;
                 }
-                if (!isset($block['EMAIL_TO_ADDRESS']) || $block['EMAIL_TO_ADDRESS'] == '') {
+                if (empty($block['EMAIL_TO_ADDRESS'])) {
                     $block['EMAIL_TO_ADDRESS'] = $to_email_address;
                 }
-                if (!isset($block['EMAIL_SUBJECT']) || $block['EMAIL_SUBJECT'] == '') {
+                if (empty($block['EMAIL_SUBJECT'])) {
                     $block['EMAIL_SUBJECT'] = $email_subject;
                 }
-                if (!isset($block['EMAIL_FROM_NAME']) || $block['EMAIL_FROM_NAME'] == '') {
+                if (empty($block['EMAIL_FROM_NAME'])) {
                     $block['EMAIL_FROM_NAME'] = $from_email_name;
                 }
-                if (!isset($block['EMAIL_FROM_ADDRESS']) || $block['EMAIL_FROM_ADDRESS'] == '') {
+                if (empty($block['EMAIL_FROM_ADDRESS'])) {
                     $block['EMAIL_FROM_ADDRESS'] = $from_email_address;
                 }
             }
             $email_html = (!is_array($block) && substr($block, 0, 6) == '<html>') ? $block : zen_build_html_email_from_template($module, $block);
-            if (!is_array($block) && ($block == '' || $block == 'none')) {
+            if (!is_array($block) && ($block === '' || $block === 'none')) {
```

Comment:
I think !is_array($block) is redundant here.

Reply:
True; that needs to be a separate PR.